src/main/java/webApp/QuestionMessage.java

```diff
@@ -9,19 +11,29 @@
     private final String question;
     private final int min;
     private final int max;
+    private final String stringAnswer;
+    private final int numberAnswer;
+    private final Collection<String> stringAnswerList;
+    private final Collection<Integer> numberAnswerList;
 
     /**
      * A numeric based question message constructor with boundaries
      * @param type  the question type
      * @param question  the question text
      * @param min   the minimum value
      * @param max   the maximum value
+     * @param stringAnswer   the answer if the type is string
+     * @param numberAnswer   the answer if the type is numerical
      */
-    public QuestionMessage(String type, String question, int min, int max) {
+    public QuestionMessage(String type, String question, int min, int max, String stringAnswer, int numberAnswer, Collection<String> stringAnswerList, Collection<Integer> numberAnswerList) {
         this.type = type;
         this.question = question;
         this.min = min;
         this.max = max;
+        this.stringAnswer = stringAnswer;
+        this.numberAnswer = numberAnswer;
+        this.stringAnswerList = stringAnswerList;
+        this.numberAnswerList = numberAnswerList;
```

Comment:
Why are these 4 required? What functionality do they add, in regards to client-side question tests?

Reply:
The QuestionMessage is overloaded to hold all question types and for multiple purposes.
The answer itself could be a string (for open ended questions) or a number (for number questions). We use a single class to transport both answers because we haven't yet figured out how to do subclassing for the purposes of json unparsing.

A similar logic applies to the answer lists.

The reason we have both answer and answer lists is so this message can go in both directions. When sending a new answer to the server, the answer is in the stringAnswer and numberAnswer fields. When retrieving previously stored survey answers, the list of previous answers are in the stringAnswerList and numberAnswerList.